packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -13,18 +13,24 @@ import { getComboboxStyles } from './getComboboxStyles';
 import { estimateSize, useComboboxFloat } from './useComboboxFloat';
 
 export type ComboboxOption<T extends string | number = string> = {
-  label: string;
+  label?: string;
   value: T;
   description?: string;
 };
 
+// TODO: It would be great if ComboboxOption["label"] was more generic so that if consumers do pass it in (for async),
+// then the onChange handler emits ComboboxOption with the label as non-undefined.
```

Comment:
You could have two types, one for options one for `onChange` where `label` would not be optional and have a type assertion for that. It would more typing but would be more correct in a sense.